database/modules/user.model.js

```diff
@@ -0,0 +1,62 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+const joi = require("joi");
+const jsonWebToken = require("jsonwebtoken");
+const config = require("../../config.env");
+const { Timestamp, ObjectId } = require("mongodb");
+
+const UserSchema = new mongoose.Schema(
+  {
+    name: { type: String, required: true },
+    email: {
+      type: String,
+      unique: true,
+      validate: {
+        validator(email) {
```

Comment:
I don't think that it's super important to validate email here. Better just validate email prior main logic